owid/owid-grapher
# 🎉 Update algolia index when publishing/updating/unpublishing gdoc posts

Resolves https://github.com/owid/owid-grapher/issues/2852

Adds a new step to the dreaded gdocs PUT route ☠️ 

**If publishing/updating:**
Update the pages index with new records

**If deleting/unpublishing:**
Delete the records from the pages index

I had to move stuff out of `indexToAlgolia.tsx` so that I could import a function without the implicit call to `indexToAlgola()` being evaluated.

Also added some comments about Fragments and https://github.com/owid/owid-grapher/issues/3426 as I encountered that annoying little gotcha.


## To test:
Set `ALGOLIA_INDEXING=true`, `ALGOLIA_ID`, `ALGOLIA_SEARCH_KEY`, `ALGOLIA_SECRET_KEY` with values from the owid-staging application on Algolia and set  in your local `.env` file

Publish a test post with at least 1500 characters (for chunking) and verify that it shows up in the owid-staging [pages index](https://dashboard.algolia.com/apps/74GKBOIDJQ/explorer/browse/pages)

Update the post to have a short amount of content and republish it. Confirm that the content is updated in the index and that records from the larger version of the post are deleted.

Change the slug of the post and republish it again. Confirm that the records are updated correctly.

Unpublish the post. Confirm that the records are deleted from the index.

Publish the post again, confirm that new records are created, and then delete the post (without unpublishing first). Confirm that the records are deleted.

Confirm that fragments don't get indexed but topic pages, linear topic pages, about pages, and articles do.
